File: src/main/java/dev/zprestige/fire/module/player/noentitytrace/MouseOverListener.java
package dev.zprestige.fire.module.player.noentitytrace;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.MouseOverEvent;
import net.minecraft.init.Items;
import net.minecraft.item.Item;

public class MouseOverListener extends EventListener<MouseOverEvent, NoEntityTrace> {

    public MouseOverListener(final NoEntityTrace noEntityTrace){
        super(MouseOverEvent.class, noEntityTrace);
    }

    @Override
    public void invoke(final Object object){
        final MouseOverEvent event = (MouseOverEvent) object;
        final Item item = mc.player.getHeldItemMainhand().getItem();
        if ((item.equals(Items.DIAMOND_PICKAXE) && module.pickaxe.GetSwitch()) || (item.equals(Items.GOLDEN_APPLE) && module.gapple.GetSwitch())){
            event.setCancelled();
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/player/noentitytrace/MouseOverListener.kt
package dev.zprestige.fire.module.player.noentitytrace

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.MouseOverEvent
import net.minecraft.init.Items

class MouseOverListener(noEntityTrace: NoEntityTrace) : EventListener<MouseOverEvent, NoEntityTrace>(MouseOverEvent::class.java,
    noEntityTrace
) {

    override fun invoke(e: Any) {
        val event = e as MouseOverEvent
        val item = mc.player.heldItemMainhand.getItem()
        if (item == Items.DIAMOND_PICKAXE && module!!.pickaxe.GetSwitch() || item == Items.GOLDEN_APPLE && module!!.gapple.GetSwitch()) {
            event.setCancelled()
        }
    }
}
